### PR TITLE
buildlib/pr/io_demo: Add a check for the out-of-sequence counter.

### PR DESCRIPTION
**Pull request by alinask**
## What
Azure CI - Add a check for the out-of-sequence counter when testing iodemo with interferences.
Check that it's increased when interference is on.

## Why ?
Verify https://redmine.mellanox.com/issues/3139906 - Nvidia internal issue.

### CODE REVIEW

**Comment by alinask** [comment]
Errors from the Azure CI:

once (doesn't seem related to this PR) :
```
==============================
Client Errors:
==============================
Log: drop_56763/tag_cx6_dc/iodemo_swx-rain02_client_00.log
Error iodemo analyzer: No traffic
[1673534891.043673] [UCX-connection 0x148a1b0: #12 2.1.6.1:20000] Remote id is 18, endpoint 0x7f8f4f8b2000, local address 2.1.6.2:33368 remote address 2.1.6.1:20000

curr_ts=1673534891.0 prev_traffic_ts=1673534710.0 curr_ts - prev_traffic_ts=180.139207 threshold=180
```

twice:
```
azure_log_error 'Out-of-sequence counter was not insreased (978)'
```

While in other cases the counter did increase as expected. For example:
```
+ [[ 974 -gt 973 ]]
+ echo 'Out-of-sequence counter was inreased during iodemo with interference'
Out-of-sequence counter was inreased during iodemo with interference
```

```
+ [[ 978 -gt 974 ]]
+ echo 'Out-of-sequence counter was inreased during iodemo with interference'
Out-of-sequence counter was inreased during iodemo with interference
```

**Comment by alinask** [comment]
@yosefe Can you please take a look?
Thanks.

**Comment by yosefe** [comment]
@alinask CI is failing

**Comment by alinask** [comment]
> @alinask CI is failing

@yosefe please see why here https://github.com/openucx/ucx/pull/8813#issuecomment-1383098841
first reason doesn't seem related this PR and the second is since counters aren't increased in some of the cases - which is what this PR is verifying.

**Comment by yosefe** [comment]
> > @alinask CI is failing
> 
> @yosefe please see why here [#8813 (comment)](https://github.com/openucx/ucx/pull/8813#issuecomment-1383098841) first reason doesn't seem related this PR and the second is since counters aren't increased in some of the cases - which is what this PR is verifying.

Seems it fails with DC. Running again to make sure

**Comment by yosefe** [comment]
Seems it's real issue with DC, will check,
in addition IMO the script needs to collect counters from all devices on both machines (currently it collects from first device on one machine) and compare the total sum

**Comment by alinask** [comment]
> Seems it's real issue with DC, will check, in addition IMO the script needs to collect counters from all devices on both machines (currently it collects from first device on one machine) and compare the total sum

Since we earlier agreed on testing only one device, can we consider what you wrote here as the next stage? so that we would have something at least for now.

**Comment by yosefe** [comment]
@alinask can you pls check local_ack_timeout counter instead of out_of_sequence?
seems out_of_sequence is not increased for DC

**Comment by alinask** [comment]
> @alinask can you pls check local_ack_timeout counter instead of out_of_sequence? seems out_of_sequence is not increased for DC

@yosefe i'm assuming you meant local_ack_timeout_err (since local_ack_timeout doesn't exist)

**Comment by yosefe** [comment]
> @yosefe i'm assuming you meant local_ack_timeout_err (since local_ack_timeout doesn't exist)

Yes, right

**Comment by alinask** [comment]
@yosefe Checking local_ack_timeout_err instead of out_of_sequence helped. All the iodemo tests passed in the CI - this counter was increased. For example:
```
+ [[ 7865 -gt 7837 ]]
+ echo 'local_ack_timeout_err counter was inreased during iodemo with interference'
local_ack_timeout_err counter was inreased during iodemo with interference
```

The current failures are not in the iodemo section:
ucp_hello_world:
```
Emulating failure before receive operation on client side
```
ucx_perftest:
```
perftest.c:406  UCX  ERROR server failed. bind() failed: Address already in use
```

File: buildlib/pr/io_demo/az-stage-io-demo.yaml [hunk 1]
```diff
@@ -104,6 +114,19 @@ steps:
     fi
     analyzer_args="$analyzer_args ${{ parameters.analyzer_allow_list_args }}"
     python ${analyzer} ${analyzer_args}
+
+    if [ '${{ parameters.interference }}' == 'Yes' ] ; then
+        device_name=$(net_device_name)
+        device_port=$(net_device_port)
+        local_ack_t_err_counter_end=$(cat /sys/class/infiniband/${device_name}/ports/${device_port}/hw_counters/local_ack_timeout_err)
+        if [[ $local_ack_t_err_counter_end -gt $(local_ack_t_err_counter_start) ]] ; then
+          echo "local_ack_timeout_err counter was inreased during iodemo with interference"
```
Comment (yosefe) [inline buildlib/pr/io_demo/az-stage-io-demo.yaml:123]: typo: increased

File: buildlib/pr/io_demo/az-stage-io-demo.yaml [hunk 2]
```diff
@@ -104,6 +114,19 @@ steps:
...
+        else
+          azure_log_error "local_ack_timeout_err counter was not insreased ($(local_ack_t_err_counter_start))"
```
Comment (yosefe) [inline buildlib/pr/io_demo/az-stage-io-demo.yaml:125]: typo: increased

**Review by yosefe — DISMISSED** [review]

**Comment by yosefe** [comment]
@alinask pls squash

**Comment by alinask** [comment]
@yosefe let's merge?

**Review by yosefe — APPROVED** [review]

**Pull request merged**